includes/utils/class-amp-image-dimension-extractor.php

```diff
@@ -2,20 +2,44 @@
 
 class AMP_Image_Dimension_Extractor {
 	static $callbacks_registered = false;
+    const STATUS_FAILED_LAST_ATTEMPT = 'failed';
+    const STATUS_IMAGE_EXTRACTION_FAILED= 'failed';
 
-	static public function extract( $url ) {
+    static public function extract( $urls ) {
```

Comment:
Thanks, Mo. I began working on this this morning but had some questions I wanted to run by you.

At first I thought you meant we needed to preserve the pre-existing functionality somehow and were asking me to change the name of the modified filter and restore the original one. I wondered, though, how the original one, which took an array and a single image URL as arguments, would be invoked, since I changed the plugin to process arrays of image URLs.

It just occurred to me that you may simply be saying we need to change the name of the filter I modified so that any existing downstream hooks that rely on its original parameters/behavior do not get invoked when the filters for it are applied.

Can you please confirm whether you mean we simply need to rename it (and make corresponding changes to the hooks in the plugin that use it) or if the original filter signature/single url functionality needs to be restored as well even though the modified plugin does not invoke/use it?

Thanks,
Mike